pkg/e2e/verify/samesite_cookie.go

```diff
@@ -21,43 +20,54 @@ const (
 	supportVersion  = 46 // samesite cookie is only supported on >= v4.6.x
 )
 
-var samesiteTestName string = "[Suite: e2e] [OSD] Samesite Cookie Strict"
+var samesiteTestName string = "[Suite: informing] [OSD] Samesite Cookie Strict"
 
 var _ = ginkgo.Describe(samesiteTestName, func() {
 	h := helper.New()
 
 	ginkgo.Context("Validating samesite cookie", func() {
 
-		ginkgo.It("should be set for openshift-monitoring OSD managed routes", func() {
-			clusterVersion, majMinVersion, err := getClusterVersion(h)
-			Expect(err).NotTo(HaveOccurred(), "failed getting cluster version")
-			Expect(clusterVersion).NotTo(BeNil())
+		checkVersion := verifyVersion(h)
 
-			if majMinVersion < supportVersion {
+		ginkgo.It("should be set for openshift-monitoring OSD managed routes", func() {
+			if checkVersion() {
 				ginkgo.Skip("skipping due to unsupported cluster version. Must be >=4.6.0")
 			}
-
 			foundKey, err := managedRoutes(h, monNamespace)
 			Expect(err).NotTo(HaveOccurred(), "failed getting routes for %v", monNamespace)
 			Expect(foundKey).Should(BeTrue(), "%v namespace routes have samesite cookie set", monNamespace)
 		}, 5)
 
 		ginkgo.It("should be set for openshift-console OSD managed routes", func() {
-			clusterVersion, majMinVersion, err := getClusterVersion(h)
-			Expect(err).NotTo(HaveOccurred(), "failed getting cluster version")
-			Expect(clusterVersion).NotTo(BeNil())
-
-			if majMinVersion < supportVersion {
+			if checkVersion() {
 				ginkgo.Skip("skipping due to unsupported cluster version. Must be >=4.6.0")
 			}
-
 			foundKey, err := managedRoutes(h, conNamespace)
 			Expect(err).NotTo(HaveOccurred(), "failed getting routes for %v", conNamespace)
 			Expect(foundKey).Should(BeTrue(), "%v namespace routes have samesite cookie set", conNamespace)
 		}, 5)
 	})
 })
 
+func verifyVersion(h *helper.H) func() bool {
+	return func() bool {
+		unsupportedVersion := false
+		clusterVersionObj, currentVersion, err := h.GetClusterVersion()
+		Expect(err).NotTo(HaveOccurred(), "failed getting cluster version")
+		Expect(clusterVersionObj).NotTo(BeNil())
+
+		// Get the cluster version and slice it, then convert the major/minor version to int Ex. majMinVersion := 46
+		splitVersion := strings.Split(currentVersion, ".")
+		majMinVersion, err := strconv.Atoi(splitVersion[0] + splitVersion[1])
```

Comment:
It's always been this way, so I guess I really should have paid more attention last time, but this can panic if the current version isn't what we expect (a semver). Is there any realistic chance of that happening? Feels like we should defensively do something like:
```suggestion
		if len(splitVersion) < 2 {
			return true // not supported
		}
		majMinVersion, err := strconv.Atoi(splitVersion[0] + splitVersion[1])
```

Reply:
While I think it would be unlikely, I am new to the team in general so I am unsure if a non semver could make its way into the API. Either way, likely worth the change to add a catch. :+1: